workflows/haplotype_sampling.wdl

```diff
@@ -0,0 +1,173 @@
+version 1.0
+
+import "../tasks/bioinfo_utils.wdl" as utils
+import "../tasks/vg_map_hts.wdl" as map
+
+workflow HaplotypeSampling {
+    meta {
+        author: "Parsa Eskandar"
+        email: "seeskand@ucsc.edu"
+        description: "Create haplotype sampled graph and the indexes necessary for the vg giraffe. More information at https://github.com/vgteam/vg/wiki/Haplotype-Sampling"
+    }
+
+    parameter_meta {
+        GBZ_FILE: "Path to .gbz index file"
+        INPUT_READ_FILE_1: "Input sample 1st read pair fastq.gz"
+        INPUT_READ_FILE_2: "Input sample 2st read pair fastq.gz"
+        HAPL_FILE: "Path to .hapl file"
+        DIST_FILE: "Path to .dist file"
+        R_INDEX_FILE: "Path to .ri file"
+        KFF_FILE: "Path to .kff file"
+        IN_OUTPUT_NAME_PREFIX: "Name of the output file (Default: haplotype_sampled_graph)"
+        IN_KMER_LENGTH: "Size of kmer using for sampling (Up to 31) (Default: 29)"
+        IN_WORKING_DIRECTORY: "Path to a directory that files are written to (Default: '.')"
+        CORES: "Number of cores to use with commands. (Default: 16)"
+        WINDOW_LENGTH: "Window length used for building the minimizer index. (Default: 11)"
+        SUBCHAIN_LENGTH: "Target length (in bp) for subchains. (Default: 10000)"
+        HAPLOTYPE_NUMBER: "Number of generated synthetic haplotypes. (Default: 4)"
+        PRESENT_DISCOUNT: "Multiplicative factor for discounting scores for present kmers. (Default: 0.9)"
+        HET_ADJUST: "Additive term for adjusting scores for heterozygous kmers. (Default: 0.05)"
+        ABSENT_SCORE: "Score for absent kmers. (Default: 0.8)"
+        INCLUDE_REFERENCE: "Include reference paths and generic paths from the full graph in the sampled graph. (Default: true)"
+    }
+    input {
+        File GBZ_FILE
+        File INPUT_READ_FILE_1
+        File? INPUT_READ_FILE_2
+        File? HAPL_FILE
+        File? DIST_FILE
+        File? R_INDEX_FILE
+        File? KFF_FILE
+        String IN_OUTPUT_NAME_PREFIX = "haplotype_sampled_graph"
+        Int IN_KMER_LENGTH = 29
+        String IN_WORKING_DIRECTORY = "."
+        Int CORES = 16
+        Int WINDOW_LENGTH = 11
+        Int SUBCHAIN_LENGTH = 10000
+        Int HAPLOTYPE_NUMBER = 4
+        Float PRESENT_DISCOUNT = 0.9
+        Float HET_ADJUST = 0.05
+        Float ABSENT_SCORE = 0.8
+        Boolean INCLUDE_REFERENCE = true
+
+
+    }
+
+    String OUTPUT_NAME_PREFIX = select_first([IN_OUTPUT_NAME_PREFIX, "haplotype_sampled_graph"])
+    Int KMER_LENGTH = select_first([IN_KMER_LENGTH, 29])
+    String WORKING_DIRECTORY = select_first([IN_WORKING_DIRECTORY, "."])
```

Comment:
None of these do anything, because the input variable types are always set to something already. They don't have the `?` optional marker on the types.